xc/device_common.ml

```diff
@@ -115,6 +115,17 @@ let disconnect_path_of_device ~xs (x: device) =
 let kthread_pid_path_of_device ~xs (x: device) =
   sprintf "%s/kthread-pid" (backend_path_of_device ~xs x)
 
+(** Where linux blkback may write its thread ids if multiple queues/rings are in use *)
+let kthread_pid_paths_of_device ~xs (x: device) =
+  let re = Re.Pcre.regexp "^queue-\\d+$" in
```

Comment:
Constructing regexes can be expensive so would it make sense to move this into the else clause?

Reply:
Maybe we could declare the regex globally, that way it only gets compiled once?

Reply:
Yes, good idea.

Reply:
Updated.